File: 2.2 logaritmo/logaritmo.m
%////////////////////////////
%/Logarithmic Transformation
%//Author: Bryan Medina
%////////////////////////////

f = imread('darkNetflix.jpg');
f = rgb2gray(f);

%constante c para obtener el rango mismo rango que la imagen de entrada
c = 255/log(1 + 255);

% Calculo de la tranformacion logaritmo
g = c * log(1 + double(f));
%conversion de la escala 0-1 a 0-255 para el correcto despliegue
g = im2uint8(mat2gray(g));

% grafica que muestra el mapeo de la funcion log
x = (0: 255);
y = log(1 + double(x));


%% cosillas para graficar
figure;
plot(x, y);
title('g(x, y) = c log(1 + f(x, y))');
xlabel('brillo - imagen de entrada f(x, y)');
ylabel('brillo - imagen de entrada g(x, y)');

figure;
subplot(1, 2, 1);
imshow(f);
title ('imagen de entrada f(x, y)');

subplot(1, 2, 2);
imshow(g);
title ('imagen de salida g(x, y)');

File: 2.2 logaritmo/logaritmo.py
# ///////////////////////
# Author : Bryan Medina
# ///////////////////////

import cv2
import numpy as np
import matplotlib.pyplot as plt

sourceImg = cv2.imread('C:/Users/bryan/Desktop/image processing python/data/darkNetflix.jpg', 0)

# calculo de la constante c, para obtener el mismo rango que el de entrada (0 - 255)
c = 255/np.log1p(sourceImg.max())

# calculo del logaritmo 
outputImg = c * np.log1p(np.double(sourceImg))


x = np.array(range(256))
y = c * np.log1p(np.double(x))

plt.figure()
plt.plot(x, y)
plt.title('g(x,y) = c log(1 + f(x, y))')
plt.xlabel('brillo - f(x,y)')
plt.ylabel('brillo - g(x, y)')
plt.show()

plt.figure()
plt.subplot(121)
plt.imshow(sourceImg, cmap='gray')
plt.title('Imagen de entrada')
plt.axis('off')

plt.subplot(122)
plt.imshow(outputImg, cmap='gray')
plt.title('Transformación logaritmo')
plt.axis('off')
plt.show()
